packages/plugin-ext/src/main/browser/webview/pre/main.js

```diff
@@ -144,15 +144,40 @@
      * @param {*} [state]
      * @return {string}
      */
-    function getVsCodeApiScript(state) {
+    function getDefaultScript(state) {
         return `
         const acquireVsCodeApi = (function() {
             const originalPostMessage = window.parent.postMessage.bind(window.parent);
+            const originalConsole = {...console};
             const targetOrigin = '*';
             let acquired = false;
 
             let state = ${state ? `JSON.parse(${JSON.stringify(state)})` : undefined};
 
+            const forwardConsoleLog= (level,msg,args)=> {
+                let message, optionalParams;
+                try{
+                  if (msg) {
+                    message= JSON.stringify(msg) ?? null;
+                  }
+                  if (args){
+                    optionalParams= JSON.stringify(args) ?? null;
+                  }
+                } catch(e){
+                    // Log non serializable objects inside of view
+                    originalConsole[level](msg,args);
+                    return;
+                }
+                originalPostMessage({ command: 'onconsole', data: { level, message,optionalParams} }, targetOrigin);
+            };
+
+            console.log= (message,args)=> forwardConsoleLog('log',message,args);
+            console.info= (message,args)=> forwardConsoleLog('info',message,args);
+            console.warn= (message,args)=> forwardConsoleLog('warn',message,args);
+            console.error= (message,args)=> forwardConsoleLog('error',message,args);
+            console.debug= (message,args)=> forwardConsoleLog('debug',message,args);
+            console.trace= (message,args)=> forwardConsoleLog('trace',message,args);
```

Comment:
Minor: More formatting. And I believe the indentation size in this method is incorrect in general.
```suggestion
                try {
                  if (msg) {
                    message= JSON.stringify(msg) ?? null;
                  }
                  if (args) {
                    optionalParams = JSON.stringify(args) ?? null;
                  }
                } catch(e) {
                    // Log non serializable objects inside of view
                    originalConsole[level](msg, args);
                    return;
                }
                originalPostMessage({ command: 'onconsole', data: { level, message, optionalParams } }, targetOrigin);
            };

            console.log = (message, args) => forwardConsoleLog('log', message, args);
            console.info = (message, args) => forwardConsoleLog('info', message, args);
            console.warn = (message, args) => forwardConsoleLog('warn', message, args);
            console.error = (message, args) => forwardConsoleLog('error', message, args);
            console.debug = (message, args) => forwardConsoleLog('debug', message, args);
            console.trace = (message, args) => forwardConsoleLog('trace', message, args);
```